test/e2e/windows/volumes.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package windows
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	emptyDirVolumePath = "C:\\test-volume"
+	hostMapPath        = "C:\\tmp"
+	containerName      = "test-container"
+	volumeName         = "test-volume"
+)
+
+var (
+	image = imageutils.GetE2EImage(imageutils.Pause)
+)
+
+var _ = SIGDescribe("Windows volume mounts ", func() {
+	f := framework.NewDefaultFramework("windows-volumes")
+	var (
+		emptiDirSource = v1.VolumeSource{
+			EmptyDir: &v1.EmptyDirVolumeSource{
+				Medium: v1.StorageMediumDefault,
+			},
+		}
+
+		hostMapSource = v1.VolumeSource{
+			HostPath: &v1.HostPathVolumeSource{
+				Path: hostMapPath,
+			},
+		}
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessNodeOSDistroIs("windows")
+	})
+
+	Context("check volume mount permissions", func() {
+
+		It("container should have readOnly permissions on emptyDir", func() {
+
+			By("creating a container with readOnly permissions on emptyDir volume")
+			doReadOnlyTest(f, emptiDirSource, emptyDirVolumePath)
```

Comment:
This test seems like just a subset of the one with two containers. Maybe we don't need this case? Or else we would want to read from a non-empty volume, maybe a configmap?

Reply:
@PatrickLang deferring to you. I agree with ben, seems to me that the function that was to be tested, is tested in the case with two containers. Is it worth putting an additional check?